### PR TITLE
enhancement: nft metadata display

### LINKED ISSUE #5035: [Task]: Add NFT metadata box on Activity Details
Currently we stringify the NFT metadata object and display it in the box.
Improve this by:
- [x] Formatting it to the default JSON format (smth like `JSON.stringify(jsObj, null, "\t")` + `whitespace-pre-wrap`)
- [x] Make the content selectable, such that users can copy pieces of the info
- [x] make sure that a long JSON box doesnt break the activty details

### PR DESCRIPTION
**Pull request by jeeanribeiro**
## Summary
Enhances how NFT metadata is displayed by adding it in activity details tabs and in mint NFT confirmation popup

## Relevant Issues
closes #5035 

### Platforms
- __Desktop__
  - [ ] MacOS
  - [x] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

## Checklist
- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/components/molecules/NftActivityDetails.svelte [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
     import { Icon as IconEnum } from '@lib/auxiliary/icon'
 
-    export let activity: Activity
+    export let activity: Partial<Activity> = {}
```
Comment (MarkNerdi) [inline packages/shared/components/molecules/NftActivityDetails.svelte:10]: Nice change with the partial, but i would remove the default, as this is always required

File: packages/shared/components/molecules/activity-info/NftMetadataInformation.svelte [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    }
+</script>
+
+{#each Object.entries(nftMetadataDetailsList) as [key, value]}
+    <KeyValueBox
+        keyText={localize(`views.collectibles.metadata.${key}`)}
+        valueText={value.data}
+        tooltipText={value.isTooltipVisible ? localize(`tooltips.transactionDetails.nftMetadata.${key}`) : undefined}
+        classes={key === 'metadata' ? 'whitespace-pre-wrap' : ''}
+        isCopyable
```
Comment (MarkNerdi) [inline packages/shared/components/molecules/activity-info/NftMetadataInformation.svelte:78]: not sure if we want every field to be copyable. Clarify that with nicole/UX

File: packages/shared/lib/core/nfts/interfaces/nft.interface.ts [hunk 3]
```diff
@@ -6,6 +6,6 @@ export interface INft {
     name: string
     issuer: AddressTypes
     metadata: string
-    parsedMetadata?: IIrc27Metadata
+    parsedMetadata?: IIrc27Metadata | string
```
Comment (MarkNerdi) [inline packages/shared/lib/core/nfts/interfaces/nft.interface.ts:9]: why can this be a string? if `metadata` is not json-parsable then `parsedMetadata` is undefined

File: packages/shared/lib/core/nfts/utils/parseNftMetadata.ts [hunk 4]
```diff
@@ -1,22 +1,110 @@
...
+        throw 'Invalid royalty value, it must be a numeric decimal representative of the percentage required ie. 0.05'
+    }
+
+    const isSumValid = Object.values(data.royalties).reduce((acc, val) => acc + val, 0) <= 1
+    if (!isSumValid) {
+        throw 'Invalid royalty value, the sum of all royalties must be less than or equal to 1'
+    }
+}
+
+function validateAttributes(data: IIrc27Metadata): void {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/nfts/utils/parseNftMetadata.ts:80]:
```suggestion
function validateAttributes(attributes: unknown): void {
```

File: packages/shared/lib/core/nfts/utils/parseNftMetadata.ts [hunk 5]
```diff
@@ -1,22 +1,110 @@
...
+    if (data.royalties) {
+        validateRoyalties(data)
+    }
+
+    if (data.attributes) {
+        validateAttributes(data)
+    }
+}
+
+function validateRoyalties(data: IIrc27Metadata): void {
```
Comment (MarkNerdi) [inline packages/shared/lib/core/nfts/utils/parseNftMetadata.ts:63]:
```suggestion
function validateRoyalties(royalties: unknown): void {
```

**Review by MarkNerdi — CHANGES_REQUESTED** [review]
Some comments to address

File: packages/shared/components/molecules/activity-info/NftMetadataInformation.svelte (hunk 2 above)
Comment (jeeanribeiro) [reply packages/shared/components/molecules/activity-info/NftMetadataInformation.svelte:78]: Clarified with Nicole, and we both agree that every field should be copyable

**Review by MarkNerdi — APPROVED** [review]
LGTM

**Pull request merged**